runtime/libc/wasi_sdk_backing.c

```diff
@@ -27,21 +27,63 @@
 #define fdatasync fsync
 #endif
 
-int main(int argc, char* argv[]) {
-    runtime_main(argc, argv);
-    printf("mem use = %d\n", (int) memory_size);
+// Code that actually runs the wasm code
+IMPORT void wasmf__start(void);
+u32 runtime_argc = 0;
+char *runtime_argv_buffer = NULL;
+u32 runtime_argv_buffer_len = 0;
+u32 *runtime_argv_buffer_offsets = NULL;
+
+void runtime_cleanup() {
+    free(runtime_argv_buffer);
+    runtime_argv_buffer_len = 0;
+    free(runtime_argv_buffer_offsets);
 }
 
-// The symbol the binary gives us to init libc
-extern void wasmf___wasm_call_ctors(void);
+int main(int argc, char* argv[]) {
+    // Set argc and argv to globals, these are later used by the WASI syscalls 
+    runtime_argc = argc;
+
+    runtime_argv_buffer_offsets = calloc(argc, sizeof(u32));
+
+    // Copy the null terminated args one after the other into a global with the format WASI calls expect
+    for (int i = 0; i < argc; i++) {
+        runtime_argv_buffer_offsets[i] = runtime_argv_buffer_len;
+        size_t arg_len = strlen(argv[i]) + 1;
+        runtime_argv_buffer = realloc(runtime_argv_buffer, runtime_argv_buffer_len + arg_len);
```

Comment:
Also would be nice to avoid allocating here

Reply:
@Others: Is there a way to avoid the dynamic allocation, though? The maximum length of the argument array is pretty stupidly large. Best we could do here (unless I'm mistaken) is to iterate through arguments, accumulating `strlen`, and do a *single* allocation instead of the repetitive `realloc`. That would be more memory and runtime effiicent.

I'm sorry @bushidocodes , but I think we do want to do this (I should have though about this when I first saw the code). We need this in sledge (to optimize for sandbox creation time), so we might as well do it right here, and replicate there. I'm OK with this is an issue TODO if we need to get this PR accepted quickly.